packages/block-library/src/image/edit.native.js

```diff
@@ -77,9 +78,17 @@ class ImageEdit extends React.Component {
 	componentDidMount() {
 		this.addMediaUploadListener();
 
-		const { attributes } = this.props;
+		const { attributes, setAttributes } = this.props;
 
 		if ( attributes.id && ! isURL( attributes.url ) ) {
+			if ( attributes.url.indexOf( 'file:' ) === 0 ) {
+				requestMediaImport( attributes.url, ( mediaId, mediaUri ) => {
+					if ( mediaUri ) {
+						this.addMediaUploadListener( );
```

Comment:
we shouldn't need to add addMediaUploadListener() call here, lately I updated this component because of a bug fix: https://github.com/wordpress-mobile/gutenberg-mobile/pull/811 it now to starts listening  at the beginning of componentDidMount and stops at componentWillUnmount

Reply:
Will update the code.